source/SystemInteractors.c

```diff
@@ -1,42 +1,125 @@
 #include "SystemInteractors.h"
-
-void FetchDrives(LPWSTR pPath)
+//Automates Taking All System-Available Drive Letter And Inputing Them To A Path Buffer.
+BOOL FetchDrives(
+	LPWSTR pPath
+)
 {
 	DWORD dwDrivesBitMask = GetLogicalDrives();//notice Dword 32bits If the function succeeds, the return value is a bitmask representing the currently available disk drives. Bit position 0 (the least-significant bit) is drive A, bit position 1 is drive B, bit position 2 is drive C, and so on.
-	if (dwDrivesBitMask == 0) return FALSE;
+	if (dwDrivesBitMask == 0) return FALSE;// If No Bit Was Set As 1 (i.e. 0b00000000 {Which Is 0}) The Function Failed And False Is Returned
 	WCHAR base_wchar = L'A';//Holding the value 0x41 (0d65) as a WideChracter (A) and using it as the base character
-	unsigned drives_index = 0;
+	unsigned drives_index = 0;//An Index Used To Track The Amount Of Drives In The System, It's Needed Because Of the Choice To Not Dynamically Allocate Memory For The First Answer As The Start Of A Working Path Follows Very Strict Rulling Which Helps Avoid Said Dynamic Allocation and TF The Responsibility To Free The Data (Helps With Recursion & Reusability).
 	for (WCHAR loop_index = 0; loop_index <= 26; loop_index++)//Casting the value of the integer held under the loopcounter to be interpeted as a wide character for human readable formatting
```

Comment:
don't use "magic numbers" - const them! for example:
```suggestion
	int MAX_NUM_OF_DRIVES = 26;
	for (WCHAR loop_index = 0; loop_index <= MAX_NUM_OF_DRIVES ; loop_index++)//Casting the value of the integer held under the loopcounter to be interpeted as a wide character for human readable formatting
```